### ISSUE TITLE
Support for Native build with Visual Studio 2015 

### ISSUE THREAD

**Issue by oyosefi** [gradle/gradle#753]
Native build on a maching with Visual Studio 2015 doesn't work (probably because CRT location changed).  I noticed this issue in JIRA and even a proposed fix (PR) a long time ago but unfortunately this still isn't fixed
### Expected Behavior

Buildling with VS toolchain should work when VS15 is instaled
### Current Behavior

doesn't find stdio.h (or something like that)
### Context

Since I have VS15 installed on my machine (for a different project) and I also need to work on a Gradle/Java project, I need to run the Gradle build in a different VM which is a pain. 
### Your Environment

- Build scan URL:

**Comment by eriwen** [comment]
@lacasseio Can we talk about prioritizing this soon? I'd like to understand the magnitude of work involved to achieve this.

**Comment by lacasseio** [comment]
We are starting the work on this now. There is some ground work that need to be done before adding this feature. We will keep you posted on this issue.

**Comment by zosrothko** [comment]
Here some comments on this issue
1/ The proposed PR fix the support of VS2015 on Win7 for the urct lib, but fails to support VS2015 on Win10 because on Win10, the Microsoft SDK locates its dependent libraies and includes in a path which is dependent of the SDK build number. For example
on Win7, one have
```
"/IC:\Program Files (x86)\Windows Kits\8.1\Include\shared"
"/IC:\Program Files (x86)\Windows Kits\8.1\Include\um"
"/IC:\Program Files (x86)\Windows Kits\8.1\Include\10.0.14393.0\ucrt"
```

while on Win10, one should have
```
"/IC:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\shared"
"/IC:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\um"
"/IC:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt"
```
The PR #704 fixes only the urct path on Win7 & Win10 while it should take also the build version number path also for 'shared' & 'um'

If you installed on your PC, the SDK 8.1 qnd the SDK 10, the fix is working because it take 'share' & 'um' from the 8.1 SDK while taking the 'urct' from the SDK 10 as
```
/IC:\Users\fandre\Documents\git\poco-gradle\Foundation\include
"/IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include"
"/IC:\Program Files (x86)\Windows Kits\8.1\Include\shared"
"/IC:\Program Files (x86)\Windows Kits\8.1\Include\um"
"/IC:\Program Files (x86)\Windows Kits\10\Include\10.0.14393.0\ucrt"
```

Please note also that this PR seems to break to gcc tool chain on Linux and I am trying to fix this but I am stuck on importing gradle into Eclipse on a Fedora platform.

**Comment by lacasseio** [comment]
Thanks for the information. We will look into modeling this better. We are very grateful for all the work you are doing on this @zosrothko.

**Comment by zosrothko** [comment]
I made a fix of the brooken toolchain selection on Linux/gcc and add it to the PR#704.

**Comment by lacasseio** [comment]
Thanks a lot @zosrothko!

**Comment by lacasseio** [comment]
PR #704 have been merged into master. Here are some the of the outstanding issues left to address:

- [ ] Update the release note
- [ ] Update the user guide
- [ ] Revert commit bef606eb594398daf44a247bc08af7dfe6b28c46
- [ ] Reenable [`NativeToolChainDiscoveryIntegrationTest.can discover tool chain in environment`](https://github.com/gradle/gradle/commit/7b69048d1699dc04b0d5558bf74f68147583f6f1#diff-d77394fecfe304a0ae931e44f0832a93R36)

Here are some of the unaddressed issues:
- gradle/gradle#892 regarding custom implementation of `snprintf` in CUnit sample integration test
- gradle/gradle#893 regarding static library mismatch at link time for Google Test and CUnit sample integration test

**Comment by lacasseio** [comment]
Release note was updated with [this commit](02168eb67da08489f71cd28e9ed20371158ec4bd).
- [ ] Code review the release note

**Comment by alexgeek** [comment]
Did this come in for 3.3? The referenced commit is a broken link.

**Comment by lacasseio** [comment]
Would you care to tell us which link you are talking about here so we can fix it? And to answer your question, Visual Studio 2015 support came in with 3.3.

**Comment by alexgeek** [comment]
> Release note was updated with [this commit](https://github.com/gradle/gradle/issues/02168eb67da08489f71cd28e9ed20371158ec4bd).

That is 404ing for me. Thanks though that's what I wanted to know.

**Comment by eriwen** [comment]
Should be
https://github.com/gradle/gradle/commit/02168eb67da08489f71cd28e9ed20371158ec4bd
